### PR TITLE
fix gen = ... gen.post() issue

### PR DESCRIPTION
**Pull request by jpihl**
@petya2164, @mortenvp what do you think about this solution?

### CODE REVIEW

File: wurf_runner.py [hunk 1]
```diff
@@ -158,6 +157,19 @@ def make_run(taskgen, run_type):
         test_files = getattr(taskgen, 'test_files', [])
         task.test_inputs = taskgen.to_nodes(test_files)
 
+        # Locate the shared libs this binary is depending on (if any).
+        # And Add these to the set of test inputs.
+        for lib in taskgen.uselib:
+            lib = taskgen.bld.task_gen_cache_names.get(lib, None)
```
Comment (mortenvp) [inline wurf_runner.py]: It's a bit complex to overwrite the lib variable.

Comment (mortenvp) [inline wurf_runner.py]: What is the type of `lib` I think that would be good to document here. Will make it easier to find the relevant code sometime in the future.

**Comment by mortenvp** [comment]
I like it a lot - it is much better. It makes sense to make copy .so files as part of the test_files attribute IMO.

**Comment by petya2164** [comment]
I will merge and release this now.

**Pull request merged**